[fix][doc] Add doc for `ackReceiptEnabled` config (#16465)
apache/pulsar: site2/docs/client-libraries-java.md

`autoAckOldestChunkedMessageOnQueueFull`|boolean|Whether to automatically acknowledge pending chunked messages when the threashold of `maxPendingChunkedMessage` is reached. If set to `false`, these messages will be redelivered by their broker. |true
`maxPendingChunkedMessage`|int| The maximum size of a queue holding pending chunked messages. When the threshold is reached, the consumer drops pending messages to optimize memory utilization.|10
`expireTimeOfIncompleteChunkedMessageMillis`|long|The time interval to expire incomplete chunks if a consumer fails to receive all the chunks in the specified time period. The default value is 1 minute. | 60000

> `ackReceiptEnabled`|boolean| If `ackReceiptEnabled` is enabled, ACK returns a receipt. The message is resent after getting the receipt.

Reviewer: > The message is resent after getting the receipt.

@coderzc - can you explain what this is supposed to mean? My understanding of `ackReceipt` is simply that the broker responds to consumer ack commands with an `ackResponse`.

Author: You are right, I will fix it.

> `ackReceiptEnabled`|boolean| If `ackReceiptEnabled` is enabled, ACK will return receipt but does not mean that the message will not be resent after get receipt.

Reviewer 2: ```suggestion
`ackReceiptEnabled`|boolean| If `ackReceiptEnabled` is enabled, ACK returns a receipt. The message is resent after getting the receipt.
```

Write in the simple present tense as much as possible if you are covering facts that were, are, and forever shall be true. https://docs.google.com/document/d/1lc5j4RtuLIzlEYCBo97AC8-U_3Erzs_lxpkDuseU0n4/edit#bookmark=id.e8uqh1awkcnp

Author: OK, thanks

Reviewer: We'll need to revisit this doc. It's not correct to say `The message is resent after getting the receipt.`.

Reviewer 3: > We'll need to revisit this doc. It's not correct to say `The message is resent after getting the receipt.`.

@coderzc @Anonymitaet Please follow up on @michaeljmarshall 's comment above. /cc @eolivelli

Reviewer 2: Thanks for your suggestions. Feel free to correct docs directly.

Author: I fixed it in PR: https://github.com/apache/pulsar/pull/16923
